lib/db.js

```diff
@@ -305,9 +330,10 @@ class LimitDBRedis extends EventEmitter {
           return callback(err);
         }
 
-        const remaining = parseInt(results[0], 10);
+        let remaining = parseInt(results[0], 10);
+        remaining = Number.isInteger(remaining) ? remaining : bucketKeyConfig.size;
```

Comment:
This calculation needed to be done here, because `remaining` is also passed into `calculateReset.`  This means that the first time a key was hit (and there were no records in Redis), `calculateReset` always came back with a bad result 😱

Reply:
😱